src/pages/studyView/StudyViewUtils.ts

```diff
@@ -63,3 +63,35 @@ export const COLORS = [
 export const NA_COLOR = '#CCCCCC'
 
 export const UNSELECTED_COLOR = '#808080'
+
+
+export function updateGeneQuery(geneQuery: string, selectedGene: string): string {
+
+    let isSelectedGenePresent = false;
+    let geneQueryLines = geneQuery.includes(':') ? geneQuery.split('\n') : _.isEmpty(geneQuery) ? [] : [geneQuery];
+    let result = _.reduce(geneQueryLines, (result: string[], queryLine) => {
+        if (queryLine.includes(':')) {
+            const gene = queryLine.substring(0, queryLine.indexOf(':')).trim();
+            const alteration = queryLine.substring(queryLine.indexOf(':') + 1).trim();
+            if (gene.toUpperCase() === selectedGene.toUpperCase()) {
+                isSelectedGenePresent = true;
+            } else {
+                result.push(gene + (alteration ? ': ' + alteration : ''));
+            }
+        } else {
+            _.forEach(queryLine.trim().split(/\s+/g), gene => {
```

Comment:
this method is doing a lot of unsafe string processing on OQL it seems, that should really be done w the OQL parser. you'll get a lot of this work done for free that way too, like pasting together the gene and the alterations and such. the OQL parser output gives you each line's query string